src/main/java/org/apache/accumulo/access/ParsedAccessExpression.java

```diff
@@ -28,16 +30,12 @@
  *
  * @since 1.0.0
  */
-public abstract class ParsedAccessExpression extends AccessExpression {
+public abstract sealed class ParsedAccessExpression extends AccessExpression
+    permits ParsedAccessExpressionImpl {
 
   private static final long serialVersionUID = 1L;
 
-  /*
-   * This is package private so that it can not be extended by classes outside of this package and
-   * create a mutable implementation. In this package all implementations that extends are
-   * immutable.
-   */
-  ParsedAccessExpression() {}
+  protected ParsedAccessExpression() {}
```

Comment:
Could probably remove this.  It only existed to prevent extension, but now that is done w/ sealed.

```suggestion
```